src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -92,7 +92,7 @@ void CARMACloudPlugin::HandleCARMARequest(tsm4Message &msg, routeable_message &r
 	}
 
 	char xml_str[10000]; 
-	sprintf(xml_str,"<?xml version=\"1.0\" encoding=\"UTF-8\"?><TrafficControlRequest><reqid>%s</reqid><reqseq>%ld</reqseq><scale>%ld</scale>%s</TrafficControlRequest>",reqid, reqseq,scale,bounds_str);
+	sprintf(xml_str,"<?xml version=\"1.0\" encoding=\"UTF-8\"?><TrafficControlRequest port=\"%s\" list=\"%s\"><reqid>%s</reqid><reqseq>%ld</reqseq><scale>%ld</scale>%s</TrafficControlRequest>",std::to_string(webport).c_str(),list_tcm.c_str(),reqid, reqseq,scale,bounds_str);

```

Comment:
Just so I understand how this works. CARMA-Cloud Plugin sends a TCR that includes the CARMA-Cloud REST API port and a bool flag for whether to send a single TCM or multiple in a list. How is the related to the uncompress functionality you added. Will a list of TCMs be sent as some kind of compressed payload?

Reply:
yes, if we specify the list = true, carma-cloud will compress/gzip the payload.